### PR TITLE
External Audit Storage: CTAs

### PR DESCRIPTION
**Pull request by mcbattirola**
Blockers:

- [x] https://github.com/gravitational/teleport.e/pull/2661 - adds page
- [x] https://github.com/gravitational/cloud/pull/6746 - add CTA event. Merged, but not deployed yet.
- [x] "Learn More" link, we currently don't have a page to redirect to (Marked as done, see note below)

Update: I've removed the "Learn More" link so we can merge this. Once we have a page to link to, I'll come back and add it.

_____

PR approved, awaiting dependencies to be merged before merging this

_____

This PR adds a new component to display a call to action to External Cloud Audits.

This will be displayed for cloud users. When the user clicks dismiss, we'll store in the local storage this info and not show it again.

Screenshot from 2023-11-17 16-29-08
Screenshot from 2023-11-17 16-28-58

Depends on https://github.com/gravitational/teleport.e/pull/2661, which implements the route we'll redirect users to and the cloud PR that adds the CTA codes https://github.com/gravitational/cloud/pull/6746.

part of https://github.com/gravitational/cloud/issues/6319

### CODE REVIEW

**Review by avatus — APPROVED** [review]
Looks good. Only note I'll throw in is if https://github.com/gravitational/teleport/pull/34734 gets merged first, you'll have to rename the `localStorage` stuff.

File: web/packages/teleport/src/Audit/Audit.tsx [hunk 1]
```diff
@@ -28,6 +28,8 @@ import {
 import useTeleport from 'teleport/useTeleport';
 import useStickyClusterId from 'teleport/useStickyClusterId';
 
+import { ExternalCloudAuditCta } from 'teleport/components/ExternalCloudAuditCta';
```
Comment (nklaassen) [inline web/packages/teleport/src/Audit/Audit.tsx:31]: nit: rename variables/types from ExternalCloudAudit -> ExternalAuditStorage for consistency

File: web/packages/teleport/src/components/ExternalAuditStorageCta/ExternalAuditStorageCta.tsx [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+import cfg from 'teleport/config';
+import { IntegrationKind } from 'teleport/services/integrations';
+import localStorage from 'teleport/services/localStorage';
+import useTeleport from 'teleport/useTeleport';
+
+import { CtaEvent } from 'teleport/services/userEvent';
+
+import { ButtonLockedFeature } from '../ButtonLockedFeature';
+
+export const Container = () => {
```
Comment (rudream) [inline web/packages/teleport/src/components/ExternalAuditStorageCta/ExternalAuditStorageCta.tsx:19]: In the new components we write, [we decided to move away from our old design pattern](https://gravitational.slack.com/archives/C02DQ1C2BMW/p1675213044004249) of having all logic handled in a `Container` which passes the props to the component.

We should move all this logic into the body of the `ExternalAuditStorageCta` component itself and scrap the `Container` wrapper, and use mocking in tests as needed to simulate the behaviour we want.

**Review by rudream — APPROVED** [review]
LGTM once comment is addressed

File: web/packages/teleport/src/components/ExternalAuditStorageCta/ExternalAuditStorageCta.tsx (hunk 2 above)
Comment (avatus) [reply web/packages/teleport/src/components/ExternalAuditStorageCta/ExternalAuditStorageCta.tsx:19]: +1, thanks for catching this

**Pull request merged**